apps/veil/src/pages/tournament/ui/lp-rewards.tsx

```diff
@@ -1,86 +1,77 @@
 import { useState } from 'react';
 import { observer } from 'mobx-react-lite';
+import { useRouter } from 'next/navigation';
 import { ChevronRight, ExternalLink } from 'lucide-react';
 import { bech32mPositionId } from '@penumbra-zone/bech32m/plpid';
+import { ValueViewComponent } from '@penumbra-zone/ui/ValueView';
 import { Pagination } from '@penumbra-zone/ui/Pagination';
 import { TableCell } from '@penumbra-zone/ui/TableCell';
-import { ValueViewComponent } from '@penumbra-zone/ui/ValueView';
-import { Button } from '@penumbra-zone/ui/Button';
 import { Density } from '@penumbra-zone/ui/Density';
-import { withdrawPositions } from '@/entities/position/api/withdraw-positions';
+import { Button } from '@penumbra-zone/ui/Button';
+import { pnum } from '@penumbra-zone/types/pnum';
+import { ValueView, Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
 import {
   useLpRewards,
   BASE_LIMIT,
   BASE_PAGE,
   LpReward,
 } from '@/pages/tournament/api/use-lp-rewards';
-import { useSortableTableHeaders } from '@/pages/tournament/ui/sortable-table-header';
+import { withdrawPositions } from '@/entities/position/api/withdraw-positions';
 import { connectionStore } from '@/shared/model/connection';
-import { useRouter } from 'next/navigation';
-import { LpRewardsSortKey } from '../server/lp-rewards';
-import { ValueView, Metadata } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
-import { pnum } from '@penumbra-zone/types/pnum';
 import { useStakingTokenMetadata } from '@/shared/api/registry';
+import { LpRewardsSortKey } from '../server/lp-rewards';
+import { useSortableTableHeaders } from './sortable-table-header';
 
-function LoadingRows() {
-  return (
-    <>
-      {new Array(5).map((_, x) => (
-        <div key={x}>
-          <TableCell cell loading>
-            null
-          </TableCell>
-          <TableCell cell loading>
-            null
-          </TableCell>
-          <TableCell cell loading>
-            null
-          </TableCell>
-          <TableCell cell loading>
-            null
-          </TableCell>
-          <TableCell cell loading>
-            null
-          </TableCell>
-        </div>
-      ))}
-    </>
-  );
-}
```

Comment:
This was explicitly factored out, because the pattern of merging in `isLoading` deep down into a component's display makes it much harder to figure out what the component is actually display as it goes through distinct states in its lifecycle.

The modification here should just be to add the case for when the rewards are known to be 0.

Reply:
How blocking this change is? Didn't hear from the web team that pattern was bad.

I suggest sticking to the [Conventional Comments](https://conventionalcomments.org/) when doing code reviews – it generally helps preventing subjective change requests and promotes discussion rather then submission to another opinion. Sticking to conventional comments, what label would you put on this comment? How blocking would you make it?

Reply:
In retrospect, I see the concern – been trying to refactor the refactored, erasing your code while doing so. Will change it tomorrow

Reply:
I would consider this a blocking suggestion. I think we've already run into a couple of bugs related to unclear handling of what is and isn't loading, so I think we should apply a soft "ratchet" of at least not regressing in terms of our handling of this.

This doesn't directly impact the current functionality, but it makes any kind of bug finding wrt the lifecycle much more tedious, in our recent experience.

Reply:
That's nice :+1: